src/coreclr/tools/superpmi/superpmi-shared/spmiutil.cpp

```diff
@@ -99,86 +101,98 @@ WCHAR* GetEnvironmentVariableWithDefaultW(const WCHAR* envVarName, const WCHAR*
     return retString;
 }
 
-#ifdef TARGET_UNIX
-// For some reason, the PAL doesn't have GetCommandLineA(). So write it.
-LPSTR GetCommandLineA()
+const char* GetEnvWithDefault(const char* envVarName, const char* defaultValue)
 {
-    LPSTR  pCmdLine  = nullptr;
-    LPWSTR pwCmdLine = GetCommandLineW();
+    // getenv isn't thread safe, but it's simple and sufficient since we are development-only tool
+    char* env = getenv(envVarName);
+    return env ? env : defaultValue;
+}
 
-    if (pwCmdLine != nullptr)
+std::string GetProcessCommandLine()
+{
+#ifdef TARGET_WINDOWS
+    return ::GetCommandLineA();
+#else
+    FILE* fp = fopen("/proc/self/cmdline", "r");
+    if (fp != NULL)
     {
-        // Convert to ASCII
+        std::string result;
+        char*       cmdLine = nullptr;
+        size_t      size    = 0;
 
-        int n = WideCharToMultiByte(CP_ACP, 0, pwCmdLine, -1, nullptr, 0, nullptr, nullptr);
-        if (n == 0)
+        while (getline(&cmdLine, &size, fp) != -1)
         {
-            LogError("MultiByteToWideChar failed %d", GetLastError());
-            return nullptr;
+            result += cmdLine;
```

Comment:
Is it desirable to concate all arguments without any separators?

This can be use some ad-hoc testing to validate that this code is producing reasonable results.

Reply:
So what do you expect this string to be when the process was invoked with `myapp arg1 arg2 arg3`?

Reply:
I see. Then using `getdelim` with `\0` would be simpler. I've tested the following code snippet to print full command line:

```cpp
#include<stdio.h>
#include<string>
#include<iostream>
int main()
{
    FILE* fp = fopen("/proc/self/cmdline", "r");
    if (fp != NULL)
    {
        std::string result;
        char*       cmdLine = nullptr;
        size_t      size    = 0;

        while (getdelim(&cmdLine, &size, '\0', fp) != -1)
        {
            // /proc/self/cmdline uses \0 as delimeter, convert it to space
            result += cmdLine;
            result += ' ';

            free(cmdLine);
            cmdLine = nullptr;
            size    = 0;
        }

        fclose(fp);
        std::cout << result << std::endl;
    }
    return 0;
}
```